subxt/src/backend/mod.rs

```diff
@@ -734,4 +872,615 @@ mod test {
+            MethodResponse::Started(MethodResponseStarted {
+                operation_id: id.to_owned(),
+                discarded_items: None,
+            })
+        }
+
+        fn operation_error(id: &str) -> FollowEvent {
+            FollowEvent::OperationError(OperationError {
+                operation_id: id.to_owned(),
+                error: "errro".to_owned(),
+            })
+        }
+
+        fn storage_done(id: &str) -> FollowEvent {
+            FollowEvent::OperationStorageDone(OperationId {
+                operation_id: id.to_owned(),
+            })
+        }
+        fn storage_result(key: &str, value: &str) -> unstable::rpc_methods::StorageResult {
+            unstable::rpc_methods::StorageResult {
+                key: Bytes(key.to_owned().into()),
+                result: rpc_methods::StorageResultType::Value(Bytes(value.to_owned().into())),
+            }
+        }
+        fn storage_items(id: &str, items: &[unstable::rpc_methods::StorageResult]) -> FollowEvent {
+            FollowEvent::OperationStorageItems(OperationStorageItems {
+                operation_id: id.to_owned(),
+                items: VecDeque::from(items.to_owned()),
+            })
+        }
+
+        fn operation_continue(id: &str) -> FollowEvent {
+            FollowEvent::OperationWaitingForContinue(OperationId {
+                operation_id: id.to_owned(),
+            })
+        }
+
+        #[tokio::test]
+        async fn storage_fetch_values_returns_stream_with_single_error() {
+            let response_data = vec![(
+                "method_response",
+                Message::Single(Ok(response_started("Id1"))),
+            )];
+            let mock_subscription_data = vec![(
+                "chainHead_v1_storage",
+                Message::Many(Ok(vec![Ok(operation_error("Id1")), Ok(FollowEvent::Stop)])),
+            )];
+            let rpc_client = setup_mock_rpc_client(false)
+                .add_method(
+                    "chainHead_v1_storage",
+                    Box::new(|data, sub, _| {
+                        Box::pin(async move {
+                            let response = data.pop("method_response".into()).single();
+                            if response.is_ok() {
+                                let item = data.pop("chainHead_v1_storage".into());
+                                if let Some(sub) = sub {
+                                    let item = item;
+                                    sub.write_delayed(item).await
+                                }
+                            }
+                            response.map(|x| RawValue::from_string(x).unwrap())
+                        })
+                    }),
+                )
+                .add_mock_data_from_iter(mock_subscription_data)
+                .add_mock_data_from_iter(response_data)
+                .build();
+
+            let (backend, mut driver): (UnstableBackend<Conf>, _) =
```

Comment:
move this unstable backend stuff to a helper function? It's used for several tests

Reply:
done